code/WorkInProgress/nadir_antenna.dm

```diff
@@ -165,34 +210,285 @@ var/global/obj/machinery/communications_dish/transception/transception_array
+			src.intcap_door_open = !src.intcap_door_open
+			src.UpdateIcon()
+		else if(!src.intcap && intcap_door_open && istype(I,/obj/item/cell))
+			boutput(user, "You install [I] into the cabinet's cell compartment.")
+			user.u_equip(I)
+			I.set_loc(src)
+			src.intcap = I
+		else
+			..(I,user)
+
+/obj/machinery/communications_dish/transception/proc/wrench_cabinet(mob/user)
+	var/cursed_check = src.repair_status - 3
+	boutput(user, "You finish [cursed_check ? "reinstalling" : "removing"] the rod retention bolts.")
+	playsound(src.loc, 'sound/items/Ratchet.ogg', 50, 1)
+	src.repair_status++
+	if(src.repair_status == TRSC_FULLREPAIR)
+		src.status &= ~BROKEN
+	src.UpdateIcon()
+
+/obj/machinery/communications_dish/transception/proc/pry_cabinet(mob/user)
+	boutput(user, "You finish prying out the damaged rods.")
+	playsound(src.loc, 'sound/items/Crowbar.ogg', 75, 1)
+	src.repair_status++
+	src.UpdateIcon()
+
+/datum/action/bar/icon/array_repair_weld
+	duration = 5 SECONDS
+	interrupt_flags = INTERRUPT_MOVE | INTERRUPT_STUNNED | INTERRUPT_ATTACKED
+	id = "array_repair_weld"
+	icon = 'icons/obj/items/tools/weldingtool.dmi'
+	icon_state = "weldingtool-on"
+	var/mob/living/user
+	var/obj/weldingtool
+	var/obj/machinery/communications_dish/transception/target
+
+	New(usermob,tool,array)
+		user = usermob
+		weldingtool = tool
+		target = array
+		..()
+
+	onUpdate()
+		..()
+		if(BOUNDS_DIST(user, target) > 0 || user == null || target == null)
+			interrupt(INTERRUPT_ALWAYS)
+			return
+
+	onStart()
+		..()
+		if(BOUNDS_DIST(user, target) > 0 || user == null || target == null)
+			interrupt(INTERRUPT_ALWAYS)
+			return
+		src.loopStart()
+
+	loopStart()
+		..()
+		if (!isweldingtool(weldingtool) || !istype(target))
+			logTheThing(LOG_DEBUG, null, "Transception array welding action bar was passed improper objects. Somehow.")
+			interrupt(INTERRUPT_ALWAYS)
+			return
+
+	onEnd()
+		if(BOUNDS_DIST(user, target) > 0 || user == null || target == null || !user.find_in_hand(weldingtool))
+			..()
+			interrupt(INTERRUPT_ALWAYS)
+			return
+
+		if(weldingtool:try_weld(user, 1))
+			target.repair_status++
+			target.UpdateIcon()
+		else
+			interrupt(INTERRUPT_ALWAYS)
+			return
+
+		if(target.repair_status > 2)
+			user.show_text("You finish welding the array cabinet.", "blue")
+			..()
+			return
+
+		src.onRestart()
 
 /obj/machinery/communications_dish/transception/update_icon()
-	if(powered())
-		var/primed_state = "allquiet"
-		if(src.primed)
-			primed_state = "glow_primed"
+	if(repair_status < TRSC_FULLREPAIR)
+		src.icon_state = "array_busted[repair_status]"
+	else
+		src.icon_state = "array[intcap_door_open ? "_panelopen" : null]"
 
-		var/image/glowy = SafeGetOverlayImage("glows", 'icons/obj/machines/transception.dmi', "glow_online")
-		glowy.plane = PLANE_ABOVE_LIGHTING
-		UpdateOverlays(glowy, "glows", 0, 1)
+	if(powered() && !(status & BROKEN))
```

Comment:
I should have put the light-documentation request down here but yeah do that

Reply:
yep